internal/vds/vds.cpp

```diff
@@ -518,6 +518,42 @@ struct vdsbuffer fetch_fence(
     return buffer;
 }
 
+struct vdsbuffer fetch_fence_metadata(
+    std::string url,
+    std::string credentials,
+    size_t npoints
+) {
+    OpenVDS::Error error;
+    OpenVDS::ScopedVDSHandle handle = OpenVDS::Open(url, credentials, error);
+
+    if(error.code != 0) {
+        throw std::runtime_error("Could not open VDS: " + error.string);
+    }
+
+    auto access = OpenVDS::GetAccessManager(handle);
+    auto const *layout = access.GetVolumeDataLayout();
+
+    if (layout->GetDimensionality() != 3) {
+        throw std::runtime_error(
+            "Unsupported VDS, expected 3 dimensions, got " +
+            std::to_string(layout->GetDimensionality())
+        );
+    }
```

Comment:
This should be extracted into a helper, as it should really be checked on every request. But it can be a TODO for later, as the same applies for other parts of these functions

Reply:
Yep, I think we agreed that this refactoring goes separately.